### PR TITLE
test(SharedMatrix): Update perf tests

### PR DESCRIPTION
**Pull request by alexvy86**
## Description

- Update the perf unit tests for SharedMatrix to be more accurate in what they do VS what they say they do.
- Decrease the maximum number of repetitions for some tests. It seems it legitimately takes a significant amount of time to complete those repetitions, to the point where the actual test runtime plus overhead of the benchmarking infrastructure sometimes hits the mocha timeout of 90s in CI.
- Exclude the output files of the memory perf tests from source control.

In the screenshot below we can see each iteration of the longer tests took ~700ms, and the perf benchmarking package does 50 iterations at a minimum to try to produce statistically significant measurements. That only accounts for 35s out of the 90s that we see the tests take in CI sometimes, but that could vary depending on the particular environment where the tests are running.

image

Addresses [AB#2963](https://dev.azure.com/fluidframework/235294da-091d-4c29-84fc-cdfc3d90890b/_workitems/edit/2963) .

### CODE REVIEW

File: packages/dds/matrix/src/test/memory/matrix.spec.ts [hunk 1]
```diff
@@ -34,95 +34,119 @@ describe("Matrix memory usage", () => {
...
 					this.localMatrix = createLocalMatrix("testLocalMatrix");
 				}
 			})(),
 		);
 
-		benchmarkMemory(
-			new (class implements IMemoryTestObject {
-				title = `Insert and remove ${x} rows`;
-				private localMatrix: SharedMatrix = createLocalMatrix("testLocalMatrix");
+		const numbersOfEntriesForTests = [100, 1000, 5000];
```
Comment (ChumpChief) [inline packages/dds/matrix/src/test/memory/matrix.spec.ts:51]: What's the motivation for doing this in 3 batches?  Is there something to observe in the small batches that we wouldn't learn in the large batches?

Comment (alexvy86) [reply packages/dds/matrix/src/test/memory/matrix.spec.ts:51]: Nothing specific I can point to, tbh; when I first wrote these (or maybe others that inspired these) my thinking was "let's look at DDS behavior under light usage, medium usage, and heavy usage; we might notice different things in each case (e.g. a memory leak that isn't obvious in the light usage case but becomes noticeable if we push the DDS further)" in an admittedly hand-wavy way.

I'll say we did end up seeing different behaviors for different batches _once_... but in that case it seems to have been an artifact of the measuring approach (which hasn't been addressed).

File: packages/dds/matrix/src/test/memory/matrix.spec.ts [hunk 2]
```diff
@@ -34,95 +34,116 @@ describe("Matrix memory usage", () => {
...
-				async run() {
-					for (let i = 0; i < x; i++) {
-						this.localMatrix.insertRows(0, 100);
-						this.localMatrix.removeRows(0, 100);
-					}
-				}
+		numbersOfEntriesForTests.forEach((x) => {
+			benchmarkMemory(
+				new (class implements IMemoryTestObject {
+					title = `Insert and remove a column ${x} times`;
```
Comment (Josmithr) [inline packages/dds/matrix/src/test/memory/matrix.spec.ts:56]: Nit: Can these `title` properties be made `readonly`?

**Review by Josmithr — APPROVED** [review]
Thank you for updating the test titles - they are **much** clearer now in my opinion.

File: packages/dds/matrix/src/test/memory/matrix.spec.ts (hunk 2 above)
Comment (alexvy86) [reply packages/dds/matrix/src/test/memory/matrix.spec.ts]: Done! The interface in @fluid-tools/benchmark should probably enforce that. I'll create a ticket for that.

**Pull request merged**